tests/FSharp.Compiler.ComponentTests/ErrorMessages/InterfaceImplInAugmentationsTests.fs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.ErrorMessages
+
+open Xunit
+open FSharp.Test.Compiler
+
+
+module InterfaceImplAugmentation =
+    [<Fact>]
+    let ``Type in non-recursive namespace show give a warning when augmented externally``() =
+        FSharp """
+namespace AugmentationsTest
+type MyCustomType<'T> = 
+    | Data of string
+    interface System.IDisposable
+
+type MyCustomType<'T> with
+    interface System.IDisposable with
+                member x.Dispose() = printfn "Finish"   """
+        |> typecheck
+        |> shouldFail
+        |> withSingleDiagnostic (Warning 69, Line 8, Col 15, Line 8, Col 33,
+                                 """Interface implementations should normally be given on the initial declaration of a type. Interface implementations in augmentations may lead to accessing static bindings before they are initialized, though only if the interface implementation is invoked during initialization of the static data, and in turn access the static data.  You may remove this warning using #nowarn "69" if you have checked this is not the case.""")
```

Comment:
Isn't it worth storing this in a const literal at the beginning of the file? so if the message is eventually adjusted there's no massive replaces across tests.

Reply:
Always felt those tests are better done in standalone F# files with baseline files of compiler output.

* It makes it easier to update the error message, compare the changes without "run tests / edit / recompile / ..." cycles, just by updating the baselines and comparing the diff through git tooling.
* looking at the F# code being checked makes more sense in standalone F# file than inside a string literal

Reply:
> Isn't it worth storing this in a const literal at the beginning of the file? so if the message is eventually adjusted there's no massive replaces across tests.

In the end changes of error messages are bulk find+replace anyway, since there are various suites of tests that use different mechanisms.

That way, it does not matter for replacing.
And for searching, it is at least one level of indirection less.

But I am rather agnostic, I also thought about referencing the text from the resource (where the compiler takes it from)....

Reply:
> Always felt those tests are better done in standalone F# files with baseline files of compiler output.
> 
> * It makes it easier to update the error message, compare the changes without "run tests / edit / recompile / ..." cycles, just by updating the baselines and comparing the diff through git tooling.
> * looking at the F# code being checked makes more sense in standalone F# file than inside a string literal

As soon as I had to look at several tests at once (trying different approaches and having different subsets failing/passing), having all the tests and assertions was a massive benefit.
For a feature having a single test, separate .fs is better if the code is non-trivial (=difficult to understand in a string-only setting).
For a suite of several smaller tests, the ability to see them (and their assertions) at glance is the selling point.